dist-build/wasm32-wasi.sh

```diff
@@ -1,37 +1,18 @@
 #! /bin/sh
 
-if [ -z "$WASI_LIBC" ]; then
-  for path in /opt/wasi-libc /opt/wasi-sysroot; do
-    if [ -d "$path" ]; then
-      export WASI_LIBC="$path"
-      break
-    fi
-  done
-fi
-if [ -z "$WASI_LIBC" ]; then
-  echo "Set WASI_LIBC to the path to the WASI libc sysroot" >&2
+if ! command -v wasienv >/dev/null; then
+  echo "Wasienv needs to be installed - https://github.com/wasienv/wasienv" >&2
   exit 1
 fi
 
-export PATH="/usr/local/opt/llvm/bin:$PATH"
-
 export PREFIX="$(pwd)/libsodium-wasm32-wasi"
 
 mkdir -p $PREFIX || exit 1
 
-export CC="clang"
-export CFLAGS="-DED25519_NONDETERMINISTIC=1 --target=wasm32-wasi --sysroot=${WASI_LIBC} -O2"
-export LDFLAGS="-s -Wl,--no-threads"
```

Comment:
`--no-threads` works around a bug in LLVM 8. Ubuntu still ships with LLVM 8 and that bug, so we still need it.